main/lsp/LSPFileUpdates.cc

```diff
@@ -148,23 +172,20 @@ LSPFileUpdates::fastPathFilesToTypecheck(const core::GlobalState &gs, const LSPC
         // This will insert two entries into `retypecheckableSymbolHashes` for each changed method, but they
         // will get deduped later.
         absl::c_set_symmetric_difference(oldRetypecheckableSymbolHashes, newRetypecheckableSymbolHashes,
-                                         std::back_inserter(changedRetypecheckableSymbolHashes));
+                                         NameHashOutputIterator(result.changedSymbolNameHashes));
 
         result.changedFiles.emplace(fref, idx);
     }
 
-    result.changedSymbolNameHashes.reserve(changedRetypecheckableSymbolHashes.size());
-    absl::c_transform(changedRetypecheckableSymbolHashes, std::back_inserter(result.changedSymbolNameHashes),
-                      [](const auto &symhash) { return symhash.nameHash; });
-    core::WithoutUniqueNameHash::sortAndDedupe(result.changedSymbolNameHashes);
-
     if (result.changedSymbolNameHashes.empty()) {
         // Optimization--skip the loop over every file in the project (`gs.getFiles()`) if
         // the set of changed symbols is empty (e.g., running a completion request inside a
         // method body)
         return result;
     }
 
+    core::WithoutUniqueNameHash::sortAndDedupe(result.changedSymbolNameHashes);
```

Comment:
This function won't empty a non-empty vector, so it's safe to move it after the check for the vector being empty.